profiles/commands.py

```diff
@@ -15,6 +20,38 @@ def _update_name_from_orcid_record(profile: Profile, orcid_record: dict) -> None
                             '{}, {}'.format(family_name, given_name))
 
 
+def _update_affiliations_from_orcid_record(profile: Profile, orcid_record: dict) -> None:
+    orcid_affiliations = orcid_record.get('activities-summary', {}).get('employments', {}).get(
+        'employment-summary', {})
+
+    for index, orcid_affiliation in enumerate(orcid_affiliations):
+        organization = orcid_affiliation['organization']
+        address = organization['address']
+
+        affiliation = Affiliation(str(orcid_affiliation['put-code']),
+                                  countries.get(address['country']), organization['name'],
+                                  date(**{k: int(v) for k, v in
+                                          orcid_affiliation['start-date'].items()}),
+                                  orcid_affiliation.get('department-name'), address.get('city'),
+                                  address.get('region'),
+                                  date(**{k: int(v) for k, v in
+                                          orcid_affiliation['end-date'].items()}, hour=23,
+                                       minute=59, second=59) if orcid_affiliation.get('end-date')
+                                  else None,
+                                  orcid_affiliation['visibility'] != VISIBILITY_PUBLIC)
+        profile.add_affiliation(affiliation, index)
+
+    for affiliation in profile.affiliations:
+        found = False
+        for orcid_affiliation in orcid_affiliations:
+            if str(orcid_affiliation['put-code']) == affiliation.id:
```

Comment:
I imagine `put-code`s are unique per-ORCID platform? All records shared a common space?

Reply:
I _think_ so; alternatively we could continue to generate a local ID and put a unique constraint on their ID + profile ID.

Reply:
https://members.orcid.org/api/tutorial/reading-xml says
```
Put codes are unique within the type of item they refer to. Using this example, only one work in the ORCID Registry will have the put code 733535, though 733535 may also used as a put code for a funding item or a keyword.
```
which is good enough for this use case